### ISSUE TITLE
Unable to restart animations after application has returned from background

### ISSUE THREAD

**Issue by adamjack** [gontovnik/DGActivityIndicatorView#12]
I am using iOS 9.2, Swift 2, Cocoapods. DGActivityIndicatorView (2.0)

I am adding animation to a login process and in some cases that goes via Facebook (which backgrounds my app.) Upon return the animation is frozen, and I cannot seem to resuscitate it (despite efforts to start or stop/start in viewDidAppear.) I've tried things I've read (on Stack Overflow) such as setting setNeedsDisplay on the animation, it's superview, the main view. No success.

It there a way to help the animation know it needs to start afresh?

**Comment by gontovnik** [comment]
Hey Adam, sorry for not replying for a long time.
Did you find a solution for your issue or you still need some help?

**Comment by susena** [comment]
Hi gontovnik,  Is it resolved?

**Comment by gontovnik** [comment]
Hey, no, I did not work on this pod for a long time.

**Comment by zfx5130** [comment]
I user it, the same problem . Unable to restart animations after application has returned from background,some type animation

**Comment by Insider77** [comment]
Same problem... :(

**Comment by Insider77** [comment]
Solved by initializing DGActivityIndicatorView in the viewDidAppear instead of viewDidLoad method

**Comment by TheLittleBoy** [comment]
I found this branch: [https://github.com/ninjaprox/DGActivityIndicatorView](https://github.com/ninjaprox/DGActivityIndicatorView)

Solved by set removedOnCompletion to NO.

**Comment by zfx5130** [comment]
@TheLittleBoy  thanks

**Comment by adamjack** [comment]
@gontovnik No apologies necessary. Thanks for sharing what you've created. Thanks to the ideas here, I'll keep them in mind. (I worked around it another way, but not a sharable code fix, I changed what I was attempting.)

**Comment by adamjack** [comment]
@gontovnik I had to return to this issue because I came upon the problem when restarting an animation within the application, not just when backgrounding. Thanks to @TheLittleBoy 'cos I finally understood his comment from above. I found that (for me) the indicator animation I am using DGActivityIndicatorBallPulseSyncAnimation needed this line added, I added it at line 31:

```
animation.removedOnCompletion = NO;
```

I don't understand the pros/cons of this, so I am not providing a patch/pull request, but it does appear in other animations. Does it make sense to add it when missing?
